detox/src/DetoxExportWrapper.js

```diff
@@ -35,7 +34,18 @@ class DetoxExportWrapper {
       Detox.none.initContext(global);
     }
 
-    this[_detox] = await DetoxExportWrapper._initializeInstance(config, params);
+    Detox.none.setError(null);
+
+    try {
+      this[_detox] = DetoxExportWrapper._createInstance(config);
+      await this[_detox].init(params);
+    } catch (err) {
+      Detox.none.setError(err);
```

Comment:
:sob: 
On the other side, we can't automatically set `this[_detox] = Detox.none` anymore. :man_shrugging: 
Is that bad, you will ask me? Yes and not exactly.

Please keep in mind that all this juggling with `Detox.none vs this[__detox]` all of a sudden is our desperate attempts to work around `jest-jasmine` bug when tests just move forward, even if detox.init() failed with a timeout.

Jest Circus solutions don't need this workaround. Mocha is fine also.
So, now, I apply another workaround. See:

```js
try {
  await adapter.beforeEach();
} catch (err) {
  await detox.cleanup();
  throw err;
}
```

By this, even after timeout error, when we recklessly continue to run, the first throwing beforeEach will result into detox.cleanup(). That's why all the consequent tests which don't plan to stop :roll_eyes:  will be failing with our Detox.none workaround.

I can't say I love all this story, but let's deprecate Jest-Jasmine support, and we might throw out some mere 150-300 lines of weird support code.